include/PreferenceList.h

```diff
@@ -27,6 +28,8 @@ class PreferenceList {
 private:
     RankType cur_rank_;        // the current unused rank
     ContainerType pref_list_;  // strictly ordered preference list
+    std::unordered_map<RankType, std::vector<PrefListElement>> tie_map_; //map to store when a particular rank has ties 
```

Comment:
Add more details to this comment. When is this data structure required, and why it is needed.
We refrain from adding type of the variable to its name, just call it ties_.